app/js/map/map.controller.js

```diff
@@ -0,0 +1,94 @@
+(function() {
+  'use strict';
+  angular.module('upcomingStops.map').controller('MapController', MapController);
+
+  MapController.$inject = ['$scope'];
+
+  function MapController($scope) {
+    var vm = this;
+    vm.map = null;
+
+    vm.latestLocation = { lat: 61.498180, lng: 23.762195 };
+
+    vm.showUserPosition = true;
+    vm.userPosMarker = null;
+
+    vm.setMap = setMap;
+    vm.getMap = getMap;
+    vm.centerToUser = centerToUser;
+
+    function setMap(map) {
+      vm.map = map;
+      initUserPositionMarker(vm.map, vm.latestLocation);
+
+      $scope.$emit('mapReady', vm.map);
+    }
+
+    function getMap() {
+      return vm.map;
+    }
+
+    function initUserPositionMarker(map, initPosition) {
+      vm.userPosMarker = map.createMarker({
+        position: initPosition,
+        icon: 'img/userpositionmarker.png'
+      });
+      vm.userPosMarker.setClickable(false);
+
+      if (vm.showUserPosition) {
+        showUserPosMarker();
+      }
+    }
+
+    function updateUserPosMarker() {
+      if (vm.userPosMarker !== null) {
+        vm.userPosMarker.setPosition(vm.latestLocation);
+      }
+    }
+
+    function showUserPosMarker() {
+      if (vm.map !== null && vm.latestLocation != null) {
+        if (vm.userPosMarker === null) {
+          initUserPositionMarker(vm.map, vm.latestLocation);
+        }
+
+        vm.map.showMarker(vm.userPosMarker);
+      }
+    }
+
+    function centerToUser() {
+      if (vm.map !== null && vm.latestLocation != null) {
+        vm.map.setZoom(15);
+        vm.map.panTo(vm.latestLocation);
+      }
+    }
+
+    llb_app.addListener('location', function(data){
```

Comment:
This is needed in the controller. Modules really aren't anything more than packages in Java.